app/src/main/java/com/keylesspalace/tusky/components/occurrence/OccurrenceActivity.kt

```diff
@@ -0,0 +1,231 @@
+        }
+    }
+
+    private inner class OccurrenceAdapter :
+        ListAdapter<OccurrenceEntity, BindingHolder<ItemOccurrenceBinding>>(OccurrenceDiffer) {
+
+        private val dateFormat = DateFormat.getDateTimeInstance(DateFormat.DEFAULT, DateFormat.SHORT)
+        private var lastAccount: AccountEntity? = null
+
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BindingHolder<ItemOccurrenceBinding> {
+            return BindingHolder(ItemOccurrenceBinding.inflate(LayoutInflater.from(parent.context), parent, false))
+        }
+
+        override fun onBindViewHolder(holder: BindingHolder<ItemOccurrenceBinding>, position: Int) {
+            val occurrence = getItem(position)
+
+            val defaultTextColor = MaterialColors.getColor(binding.root, android.R.attr.textColorPrimary)
+
+            holder.binding.what.text = occurrence.what
+            holder.binding.what.setTextColor(
+                if (occurrence.type == OccurrenceEntity.Type.CRASH) {
+                    Color.RED
+                } else {
+                    defaultTextColor
+                }
+            )
+
+            holder.binding.code.text = occurrence.code?.toString() ?: ""
+            holder.binding.code.setTextColor(
+                if (occurrence.code != null && occurrence.code > 0) {
+                    if (occurrence.code >= 400) {
+                        Color.RED
+                    } else if (occurrence.code >= 300) {
+                        Color.YELLOW
+                    } else {
+                        Color.GREEN
+                    }
+                } else {
+                    defaultTextColor
+                }
+            )
+
+            holder.binding.whenDate.text =
+                getRelativeTimeSpanString(this@OccurrenceActivity.applicationContext, occurrence.startedAt.time, System.currentTimeMillis())
+                //dateFormat.format(occurrence.startedAt)
+            // TODO or AbsoluteTimeFormatter?
+
+            // TODO how does one get the current locale /and/or format numbers here?
+            val currentLocale = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                resources.configuration.locales[0]
+            } else {
+                resources.configuration.locale
+            }
+
+            var duration = ""
+            var durationMs = 0L
+            if (occurrence.finishedAt != null) {
+                durationMs = occurrence.finishedAt.time - occurrence.startedAt.time
+                duration = getDurationStringAllowMillis(currentLocale, durationMs)
+            }
+            holder.binding.duration.text = duration
+            holder.binding.duration.setTextColor(
+                if (durationMs >= 1000) {
+                    Color.MAGENTA
+                } else if (durationMs >= 400) {
+                    // TODO these colors are a bit of a problem: e. g. on light mode yellow is hardly visible and green is difficult
```

Comment:
make them resources and use different ones for light and dark mode

Reply:
Done.